WordPress/src/main/java/org/wordpress/android/ui/blaze/BlazeFeatureUtils.kt

```diff
@@ -77,4 +74,78 @@ class BlazeFeatureUtils @Inject constructor(
             mapOf(SOURCE to blazeFlowSource.trackingName)
         )
     }
+
+    fun trackFlowError(blazeFlowSource: BlazeFlowSource, blazeFlowStep: BlazeFlowStep) {
+        analyticsTrackerWrapper.track(
+            AnalyticsTracker.Stat.BLAZE_FLOW_ERROR,
+            mapOf(
+                SOURCE to blazeFlowSource.trackingName, CURRENT_STEP to blazeFlowStep.trackingName
+            )
+        )
+    }
+
+    fun trackFlowCanceled(blazeFlowSource: BlazeFlowSource, blazeFlowStep: BlazeFlowStep) {
+        analyticsTrackerWrapper.track(
+            AnalyticsTracker.Stat.BLAZE_FLOW_CANCELED,
+            mapOf(
+                SOURCE to blazeFlowSource.trackingName, CURRENT_STEP to blazeFlowStep.trackingName
+            )
+        )
+    }
+
+    fun trackBlazeFlowStarted(blazeFlowSource: BlazeFlowSource) {
+        analyticsTrackerWrapper.track(
+            AnalyticsTracker.Stat.BLAZE_FLOW_STARTED, mapOf(SOURCE to blazeFlowSource.trackingName)
+        )
+    }
+
+    fun trackBlazeFlowCompleted(blazeFlowSource: BlazeFlowSource) {
+        analyticsTrackerWrapper.track(
+            AnalyticsTracker.Stat.BLAZE_FLOW_COMPLETED, mapOf(SOURCE to blazeFlowSource.trackingName)
+        )
+    }
+
+    @Suppress("ReturnCount")
+    fun extractCurrentStep(url: String?): BlazeFlowStep {
```

Comment:
❓ Any particular reason why this logic is extracted to the utils class? It looks like the logic is related to the UI more. 

💡 I would suggest moving this to the `BlazeWebViewViewModel`. But leave it to you if you want to keep it here.

Reply:
@AjeshRPai - I had planned on using it elsewhere, but it makes sense to move it over to the `BlazeWebViewViewModel`. Thanks for the suggestion. In commit [215b633](https://github.com/wordpress-mobile/WordPress-Android/pull/18013/commits/215b633ac9387b101d2ad225472c779e144e16ae)